app/src/main/kotlin/no/nav/omsorgsdager/vedtak/dto/VedtakNøkkelinformasjon.kt

```diff
@@ -10,20 +10,22 @@ internal object VedtakNøkkelinformasjon {
         val status: VedtakStatus,
         val potensielleStatuser: Map<String, Any>,
         val uløsteBehov: Map<String, Any>) {
-        internal constructor(
-            vedtak: Vedtak,
-            behov: Behov) : this(
-            status = vedtak.status,
-            potensielleStatuser = when (vedtak.status) {
-                VedtakStatus.FORESLÅTT -> when {
-                    behov.løsteBehov.kanInnvilges() -> setOf(
-                        VedtakStatus.INNVILGET, VedtakStatus.AVSLÅTT, VedtakStatus.FORKASTET
-                    )
-                    else -> setOf(VedtakStatus.AVSLÅTT, VedtakStatus.FORKASTET)
-                }
-                else -> emptySet()
-            }.associateBy { it.name }.mapValues { Any() },
-            uløsteBehov = behov.uløsteBehov.associateBy { it.navn }.mapValues { Any() }
+        internal constructor(vedtakOgBehov: Pair<Vedtak, Behov>) : this(
+            status = vedtakOgBehov.first.status,
+            potensielleStatuser = vedtakOgBehov.potensielleStatuser()
+                .associateBy { it.name }.mapValues { Any() },
+            uløsteBehov = vedtakOgBehov.second.uløsteBehov
+                .associateBy { it.navn }.mapValues { Any() }
         )
     }
+
+    internal fun Pair<Vedtak, Behov>.potensielleStatuser() = when (first.status) {
+        VedtakStatus.FORESLÅTT -> when {
+            second.løsteBehov.kanInnvilges() -> setOf(
+                VedtakStatus.INNVILGET, VedtakStatus.AVSLÅTT, VedtakStatus.FORKASTET
+            )
+            else -> setOf(VedtakStatus.AVSLÅTT, VedtakStatus.FORKASTET)
+        }
+        else -> emptySet()
+    }
```

Comment:
Ser vi mistet dimensjonen om at det ikke må være noen uløste behov for å innvilge..

Så gjør denne private og behold bruken av den ovenfor som den er - og heller lage 3 nye internal funksjoner her som er de vi bruker i Routen;

`internal fun Pair<Vedtak, Behov>.kanInnvilges() = potensielleStatuser().contains(VedtakStatus.INNVILGET) && second.uløsteBehov.isEmpty()`

`internal fun Pair<Vedtak, Behov>.kanAvslås() = potensielleStatuser().contains(VedtakStatus.AVSLÅTT)`

`internal fun Pair<Vedtak, Behov>.kanForkastes() = potensielleStatuser().contains(VedtakStatus.FORKASTET)`

Reply:
Fikset.